city_scrapers/spiders/chi_state_university.py

```diff
@@ -0,0 +1,151 @@
+                        source=self._parse_source(response),
+                    )
+                    meeting["status"] = self._get_status(meeting)
+                    meeting["id"] = self._get_id(meeting)
+
+                    yield meeting
+
+    def _parse_description(self, item):
+        parts = item.xpath(".//text()").extract()
+        description = "".join([str(x) for x in parts])
+        description = (
+            description.replace("\xa0", " ")
+            .replace("\n", " ")
+            .replace("                                     ", " ")
+            .replace("\t", " ")
+            .strip()
+        )
+        return description
+
+    def _parse_classification(self, i):
+        classifications = [BOARD, BOARD, COMMITTEE]
+        return classifications[i]
+
+    def _parse_start(self, item):
+        text = item.xpath(".//text()").extract()[0]
+        text = text.replace("\xa0", "").strip()
+
+        today = datetime.now()
+        yearMatch = today.year
+        monthMatch = today.month
+        dayMatch = today.day
+        hourMatch = 0
+        minuteMatch = 0
+
+        try:
+            yearMatch = int(re.search(r"\d{4}", text).group(0))
+        except AttributeError:
+            pass
+
+        try:
+            dayMatch = int(re.search(r"\d{1,2}", text).group(0))
+        except AttributeError:
+            pass
+
+        try:
+            months = {m.lower() for m in month_name[1:]}
+            monthMatch = next(
+                (word for word in text.split() if word.lower() in months), None
+            )
+            monthMatch = datetime.strptime(monthMatch, "%B").month
+        except AttributeError:
+            pass
+
+        try:
+            minuteMatch = re.search(r":([0-5][0-9])", text).group(0)
+            minuteMatch = int(minuteMatch.replace(":", ""))
+        except AttributeError:
+            pass
+
+        try:
+            hourMatch = re.search(r"(1[0-2]|0?[1-9]):", text).group(0)
+            hourMatch = int(hourMatch.replace(":", ""))
+        except AttributeError:
+            pass
+
+        if "p.m." in text:
+            hourMatch += 12
+
+        return datetime(yearMatch, monthMatch, dayMatch, hourMatch, minuteMatch)
+
+    def _parse_time_notes(self, item):
+        text = " ".join(item.xpath(".//text()").extract())
+        text = text.replace("\xa0", "").strip()
+
+        notes = ""
+        if "reschedule" in text.lower() or "postpone" in text.lower():
+            notes = text
+        if "tbd" in text.lower():
+            notes = "Meeting time TBD"
+        return notes
+
+    def _parse_location(self, item):
+
+        return {
+            "address": "9501 S. King Drive Chicago, IL 60628",
+            "name": "Room 15, 4th Floor, Gwendolyn Brooks Library Auditorium",
+        }
+
+    def _parse_links(self, item):
+
+        try:
+            link = item.xpath(".//a").attrib["href"]
+        except KeyError:
+            link = ""
+        title = ""
```

Comment:
We'll want the title to default to the link text here if it doesn't match any of the options

Reply:
To confirm, this would be the title of all meeting types to "Meeting Dates"?
Is it still ok to base the meeting classification on the tabs?

Reply:
Sorry, I think I might have misunderstood. I'm only talking about the link titles here, but for the comment above it would be great if we could pull the text from the tab rather than base it on the index

Reply:
Sorry my mistake, I was meant to comment on your above comment about title. I see what you mean. I tried to do this but found it difficult because those titles are written inside `<label>` objects that sit outside of the tab divs themselves.

For example
```
<input id="tabthree" type="radio" name="itd-tabs"></input> <label class="itd-label" for="tabthree">Standing Committee Meeting</label>
<div class="itd-tab">
```

This is why the titles were collected separately. What do you think?

Reply:
Gotcha, thanks for clarifying! It looks like with the way they have it set up the tab div has to be the immediate sibling of the label. You should be able to get this working by iterating over the labels instead, and inside the loop you can parse the title from the label and then access the tab with the `following-sibling` selector in XPath